staging/src/k8s.io/pod-security-admission/admission/admission_test.go

```diff
@@ -457,3 +465,201 @@ func TestValidateNamespace(t *testing.T) {
 		})
 	}
 }
+
+func TestValidatePodController(t *testing.T) {
+	objMetadata := metav1.ObjectMeta{Name: "testName", Namespace: "testNS", Labels: map[string]string{"foo": "bar"}}
+	// One of the pod-template objects
+	oldDeploy := appsv1.Deployment{
+		ObjectMeta: objMetadata,
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: objMetadata,
+				Spec: corev1.PodSpec{
+					RuntimeClassName: pointer.String("containerd"),
+				},
+			},
+		},
+	}
+	newDeploy := appsv1.Deployment{
+		ObjectMeta: objMetadata,
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: objMetadata,
+				Spec: corev1.PodSpec{
+					SecurityContext: &corev1.PodSecurityContext{
+						// out of allowed sysctls to return auditAnnotation or warning
+						Sysctls: []corev1.Sysctl{{Name: "unkown", Value: "unkown"}},
+					},
+					RuntimeClassName: pointer.String("containerd"),
+				},
+			},
+		},
+	}
+
+	testCases := []struct {
+		desc                 string
+		exemptNamespaces     []string
+		exemptRuntimeClasses []string
+		exemptUsers          []string
+		// request subresource
+		subresource string
+		// labels for the  namespace
+		nsLabels map[string]string
+		// for create
+		newObject runtime.Object
+		// for update
+		oldObject runtime.Object
+		gvr       schema.GroupVersionResource
+
+		expectAllowed          bool
+		expectWarnings         []string
+		expectAuditAnnotations map[string]string
+	}{
+		{
+			desc:          "status updates don't produce warnings",
```

Comment:
the tests that aren't setting nsLabels aren't really exercising subresource or namespace exemption... 

I would suggest setting nsLabels to this for all tests, rather than making it a parameter in the test struct:

```
map[string]string{
  api.EnforceLevelLabel: string(api.LevelBaseline),
  api.WarnLevelLabel: string(api.LevelBaseline),
  api.AuditLevelLabel: string(api.LevelBaseline),
},
```

then you can make sure subresources and exempt namespaces are ignored, and that the enforce label doesn't apply to the controller object, only warn and audit

Reply:
Okay, to ensure that even if there is a baseline strategy, it will be exempted or normal.
Done.